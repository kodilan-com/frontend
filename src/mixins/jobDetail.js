import formatDate from '../utils/date';
import { JOB_TYPE_MAP } from '../store/constants';

export default {
  props: {
    post: {
      type: Object,
      required: true,
    },
  },
  computed: {
    detailsUrl() {
      return `/ilanlar/${this.post.slug}`;
    },
    company() {
      return this.post.company;
    },
    timeago() {
      return formatDate(this.post.updated_at);
    },
    jobType() {
      return JOB_TYPE_MAP[this.post.type] || '';
    },
    postTypeClassName() {
      const CLASS_MAP = {
        0: 'full-time',
        1: 'full-time',
        2: 'part-time',
        3: 'internship',
        4: 'freelance',
      };

      return {
        [CLASS_MAP[this.post.type]]: true,
      };
    },
  },
};
