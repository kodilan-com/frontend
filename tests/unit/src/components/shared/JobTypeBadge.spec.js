import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import JobTypeBadge from '@/components/shared/JobTypeBadge';
import jobDetail from '@/mixins/jobDetail';

jest.mock('@/mixins/jobDetail', () => ({
  props: {
    post: {
      type: Object,
    },
  },
  computed: {
    jobType: jest.fn(),
    postTypeLink: () => 'test-link',
    postTypeClassName: () => 'test-class-name',
  },
}));

describe('JobTypeBadge.vue', () => {
  let wrapper;
  const renderComponent = (type) => {
    wrapper = shallowMount(JobTypeBadge, {
      propsData: {
        post: {
          type: type || '1',
        },
      },
      stubs: {
        RouterLink: RouterLinkStub,
      },
    });
  };

  beforeEach(() => {
    jobDetail.computed.jobType.mockReturnValue('test');

    renderComponent();
  });

  afterEach(() => {
    jest.restoreAllMocks();

    wrapper.destroy();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot, if there jobType is not falsy', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should return undefined, if there jobType is falsy', () => {
    jobDetail.computed.jobType.mockReturnValue('');

    renderComponent();

    expect(wrapper.html()).toBe(undefined);
  });

  it('should render correct post type link', () => {
    expect(wrapper.find('a.tag-post-type').props('to')).toBe('/ilan-ara/tam-zamanli');

    renderComponent('2');

    expect(wrapper.find('a.tag-post-type').props('to')).toBe('/ilan-ara/yari-zamanli');

    renderComponent('3');

    expect(wrapper.find('a.tag-post-type').props('to')).toBe('/ilan-ara/stajyer');

    renderComponent('4');

    expect(wrapper.find('a.tag-post-type').props('to')).toBe('/ilan-ara/freelance');
  });
});
