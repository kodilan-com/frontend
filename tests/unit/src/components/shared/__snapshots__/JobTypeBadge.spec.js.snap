// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobTypeBadge.vue should match snapshot, if there jobType is not falsy 1`] = `
<span class="listing-type tag-post-type test-class-name"><a itemprop="employmentType" class="tag-post-type">
    test
  </a></span>
`;
